File: matriMult_int_jki.c
#include <time.h>
#include <stdio.h>
#include <stdlib.h>
int main() {
    int n = 512;

        int i, j, k;
        int a[n][n], b[n][n], M[n][n] ;

        for(i = 0; i < n; i++) {
            for(j = 0; j < n; j++) {
                a[i][j] = rand();
                b[i][j] = rand();
            }
        }
        double time;
        time = clock();        
        for(j=0;j<n;j++) {    
            for(k=0;k<n;k++) {    
                M[i][j]=0;    
                for(i=0; i<n; i++){    
                    M[i][j]+=a[i][k]*b[k][j];    
                }    
            }    
        }    
        time = clock() - time;
        time = time/CLOCKS_PER_SEC;
        printf("\n");
        printf("In C with combination j k i\n");
        printf("Execution time for (N = %d) is %f seconds.\n",n, time);
        printf("\n");
    
	return 0;
}

File: matriMult_int_jki.py
import numpy
import time

n= 512
a = numpy.random.randint(50, size=(n,n))
b = numpy.random.randint(50, size=(n,n))
# print(a)
# print(b)
M = a*0
t = time.time()
for j in range (n) :
    for k in range(n):
        for i in range (n):
            M[i][j] +=(a[i][k] * b[k][j])
            
#M = a.dot(b)
# print (M)
print()
print("In Python with combination j k i")
print (n,"n : Execution time in seconds : ",time.time() - t)
print()